pkg/variablevalidator/variablevalidator.go

```diff
@@ -86,16 +86,25 @@ func (v *validatorVisitor) EnterVariableDefinition(ref int) {
 		v.StopWithInternalErr(err)
 		return
 	}
-	if err := schemaValidator.Validate(context.Background(), variable); err != nil {
-		message := err.Error()
-		var validationErr *jsonschema.ValidationError
-		if errors.As(err, &validationErr) && len(validationErr.Causes) > 0 {
-			message = validationErr.Causes[0].Message
-		}
-
-		v.StopWithExternalErr(operationreport.ErrVariableValidationFailed(variableName, message, v.operation.VariableDefinitions[ref].VariableValue.Position))
-		return
-	}
+ 	if err := schemaValidator.Validate(context.Background(), variable); err != nil {
+ 		message := err.Error()
+ 		var validationErr *jsonschema.ValidationError
+ 		if errors.As(err, &validationErr) && len(validationErr.Causes) > 0 {
+ 			message = validationErr.Causes[0].Message
+ 		}
+ 
+   		if strings.Contains(message, "required") {
```

Comment:
Can we remove this code repetition? Seems like we have similar logic in pkg/graphql/errors.go?